test/tests/functional/pbs_wean_udp.py

```diff
@@ -0,0 +1,123 @@
+        return rid
+
+    def test_comm_with_mom(self):
+        """
+        Test the installation of communication daemon during PBS installation.
+        """
+        msg = "Need 2 moms as input. use -pmoms=<m1>:<m2>"
+        if len(self.moms) != 2:
+            self.skip_test(reason=msg)
+        (self.momA, self.momB) = self.moms.values()
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        log_msg = ["TPP initialization done",
+                   "Single pbs_comm configured, TPP Fault tolerant mode disabled",
+                   "Connected to pbs_comm %s.*:17001" % self.server.shortname]
+        for msg in log_msg:
+            self.server.log_match(msg, regexp=True)
+            self.scheduler.log_match(msg, regexp=True)
+            self.momA.log_match(msg, regexp=True)
+            self.momB.log_match(msg, regexp=True)
+        server_ip = socket.gethostbyname(self.server.hostname)
+        msg = "Registering address %s:15001 to pbs_comm" % server_ip
+        self.server.log_match(msg)
+        msg = "Registering address %s:15004 to pbs_comm" % server_ip
+        self.scheduler.log_match(msg)
+        for mom_name in self.moms.values():
+            ip = socket.gethostbyname(mom_name.shortname) 
+            msg1 = "Registering address %s:15003 to pbs_comm" % ip
+            msg2 = "Leaf registered address %s:15003" % ip
+            mom_name.log_match(msg1)
+            self.comm.log_match(msg2)
+        # Submit job
+        jid = self.submit_job(exp_attrib={'job_state': 'R'})
+        self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
+        self.server.log_match("%s;Exit_status=0" % jid)
+        # Submit multi-chunk job
+        set_attrib = {'Resource_List.select': '2:ncpus=1'}
+        jid = self.submit_job(set_attrib, exp_attrib={'job_state': 'R'})
+        self.server.expect(JOB, 'queue', id=jid, op=UNSET, offset=10)
+        self.server.log_match("%s;Exit_status=0" % jid)
+        # Submit interactive job
+        set_attrib = {ATTR_inter: ''}
+        jid = self.submit_job(set_attrib, interactive=True)
+        # Submit reservation
+        set_attrib = {'Resource_List.select': '1:ncpus=1',
+                      'reserve_start': int(time.time() + 5),
+                      'reserve_end': int(time.time() + 120)}
+        exp_attrib = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        rid = self.submit_resv(set_attrib, exp_attrib)
+        resv_que = rid.split('.')[0] 
+        # Submit job into reservation
+        set_attrib = {ATTR_q: resv_que}
+        jid = self.submit_job(set_attrib, sleep=60)
+        # Wait for reservation to start
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|2')}
+        self.server.expect(RESV, a, rid, offset=5)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.server.delete(rid)
+        # wait for sometime for the job in reservation queue to get delete
+        time.sleep(5)
```

Comment:
I literally don't like this way of waiting for the job to vanish, what if the machine is too slow? will this time sufficient?
instead, we should use server.expect() with queue attr unset on job or check job end message in server/accounting log